libqtile/core/manager.py

```diff
@@ -660,6 +671,8 @@ def manage(self, win: base.WindowType) -> None:
             if not win.group and self.current_screen.group:
                 self.current_screen.group.add(win, focus=win.can_steal_focus)
         self.core.update_client_list(self.windows_map)
+        self.core.update_client_stack()
```

Comment:
I'm thinking it might be good to move these into the X Core as well, and it seems like that would be straight forward, and I *think* without unexpected side effects.

In this case, `Qtile.manage()` is called by the Core, and we can move these two `upate_client_X` calls to after `manage()` returns to the Core. Functionally the only difference is that the client_managed hook is moved to before the client list and stack are updated. I suspect that might be fine?

Then only other place these functions are used is the inverse: in `Qtile.unmanage()`. In that case we could either move it to after the call returns, moving the client_killed hook to before the client list/stack update, which might be fine. Or alternatively the Core can copy `self.windows_map`, `del self.windows_map[wid]` from the copy, and update the list/stick before calling `Qtile.unmanage`. That one might be safer because who knows what users are doing in client_killed hooks.

If we make these changes we can also drop `update_client_list` and `update_client_stack` from the base `Core` class.

Reply:
I've moved these to the x11 core and subscribed to `client_managed` and `client_killed` hooks to call the methods.